libs/parallelism/algorithms/include/hpx/parallel/algorithms/rotate.hpp

```diff
@@ -76,18 +246,18 @@ namespace hpx { namespace parallel { inline namespace v1 {
             {
             }
 
-            template <typename ExPolicy, typename InIter>
+            template <typename ExPolicy, typename InIter, typename Sent>
             static IterPair sequential(
-                ExPolicy, InIter first, InIter new_first, InIter last)
+                ExPolicy, InIter first, InIter new_first, Sent last)
             {
                 return detail::sequential_rotate(first, new_first, last);
             }
 
-            template <typename ExPolicy, typename FwdIter>
+            template <typename ExPolicy, typename FwdIter, typename Sent>
             static typename util::detail::algorithm_result<ExPolicy,
                 IterPair>::type
-            parallel(ExPolicy&& policy, FwdIter first, FwdIter new_first,
-                FwdIter last)
+            parallel(
+                ExPolicy&& policy, FwdIter first, FwdIter new_first, Sent last)
             {
                 return util::detail::algorithm_result<ExPolicy, IterPair>::get(
```

Comment:
the second parameter of algorithm_result  'IterPair' is fine here?

Reply:
Yes, if you look at line 242, the iterpair we use here is the one being used. When calling the function in the tag dispatch overloads    
```
parallel::v1::detail::rotate<
hpx::parallel::util::in_out_result<FwdIter, Sent>>()
.call(hpx::execution::seq, first, new_first, last)
```
Here IterPair would refer to hpx::parallel::util::in_out_result<FwdIter, Sent>. So IterPair is just used to refer to the return type.

Reply:
much clear!